### ISSUE TITLE
Provide this package as an ES Module (ESM)

### ISSUE THREAD

**Issue by greengiraffe** [felixmosh/i18next-hmr#148]
EDIT: this does support ESM, as correctly stated here: https://github.com/felixmosh/i18next-hmr/issues/148#issuecomment-1725679735

**Is your feature request related to a problem? Please describe.**

I'd like to use this plugin in an ESM-based environment, which doesn't seem to be possible at the moment since the lib only exports CJS modules (unless I'm missing something). Since i18next itself is provided as CJS _and_ ES, it's possible to write pure ESM-based configs in which this plugin cannot be used.

**Describe the solution you'd like**

If possible provide an ESM compatible export in addition to the existing CJS export.

**Describe alternatives you've considered**

I couldn't find any alternatives.

**Additional context**
–

**Comment by felixmosh** [comment]
Hi @greengiraffe, as far as I know, [esm allows](https://adamcoster.com/blog/commonjs-and-esm-importexport-compatibility-examples#how-to-import-commonjs-cjs-into-esm) to import a cjs modules, so what is the issue that you are getting?

Which build env do you use? webpack or vite?

**Comment by greengiraffe** [comment]
Hi @felixmosh thanks for your quick reply. You're absolutely right, ESM doesn't seem to be the issue, I was just stupid. My TS config was configured incorrectly, the `moduleResolution` was set to `node` instead of `nodenext`...

I cannot get the HMR to work properly though. I'm using Next.js with react-i18next instead of next-i18next since I'm exporting an SPA (no SSR/SSG). I'm basically using the config from the Next.js example to set up the HMRPlugin in the i18n config as well as adding it to the next.config.js.

This setup is possibly a bit obscure, so I don't expect to find a fix, but maybe it's obvious to you from the snippets provided below. I get the "[I18NextHMR] Client HMR has started" log but when changing language files, nothing is reloaded.

The versions I'm using are:

i18next-hmr: 3.0.2
next.js: 13.4.19
i18next: 23.5.1
i18next-http-backend: 2.2.2

i18n.ts:

```ts
import i18n from 'i18next';
import LanguageDetector, {DetectorOptions} from 'i18next-browser-languagedetector';
import {HMRPlugin} from 'i18next-hmr/plugin';
import Backend, {HttpBackendOptions} from 'i18next-http-backend';
import {initReactI18next} from 'react-i18next';

const i18nInstance = i18n.use(Backend).use(LanguageDetector).use(initReactI18next);

if (process.env.NODE_ENV !== 'production') {
  i18nInstance.use(
    new HMRPlugin({
      webpack: {
        client: true,
      },
    }),
  );
}

i18n.init<HttpBackendOptions & DetectorOptions>({
  backend: {
    loadPath: '/static/locales/{{lng}}/{{ns}}.json',
  },
  react: {
    useSuspense: false,
  },
});

export {i18n};

// reexport functions & types from react-i18next
export {useTranslation, Trans, I18nextProvider} from 'react-i18next';
export type {TFunction} from 'i18next';
```

next.config.js snippet:

```ts
 webpack(config, {isServer}) {
    if (!isServer && config.mode === 'development') {
      console.log(path.resolve(__dirname, 'public/static/locales'));
      const {I18NextHMRPlugin} = require('i18next-hmr/webpack');
      config.plugins.push(
        new I18NextHMRPlugin({
          localesDir: path.resolve(__dirname, 'public/static/locales'),
        }),
      );
    }
    return config;
}
```

**Comment by felixmosh** [comment]
Your code looks ok to me, it should work as it is only on the client.
When you change some json file, do you see any line change at the Next.js terminal output?
Do u see any console message in the browser? (except the one that tells you that "client HMR has started")

**Comment by greengiraffe** [comment]
No console output, neither in the browser nor the terminal.

I've created a sandbox that reproduces the issue here: https://codesandbox.io/p/sandbox/exciting-bessie-yf9hgt

**Comment by felixmosh** [comment]
The only piece that is not properly tested failed :]
Can you test `v3.0.3` (make sure you've removed the `.next` folder before)

**Comment by greengiraffe** [comment]
It works now using `v3.0.3`. Thanks for your quick response! :)

As mentioned above, I had to change the values for "module" and "moduleResolution" in my tsconfig from `esnext` & `node` to `NodeNext` in order to import the HMRPlugin. 

Also, a side note regarding an issue that I assume is something on Codesandbox' end: I've updated the sandbox that I've linked above – there it doesn't seem to work fully. The terminal and browser console log the updated translation file, but the page isn't hot-reloaded. Also, when editing and saving the `pages/index.ts`, the in-app-browser seems stuck in an infinite reloading loop.
